### PR TITLE
error behavior in gatsby for non-existent dataset urls

### PR DESCRIPTION
**Pull request by eharkins**
these changes allow you to go to
- `/ncov/*`
- `/sars-cov-2/*`
and see the sars-cov-2 page.

If you go to `/ncov/*` it's assumed
you want a dataset but if you land on
the sars-cov-2 gatsby page then that
dataset doesn't exist (otherwise you
would get routed to auspice) so we
display a banner letting you know that
the dataset doesn't exist but don't
wipe the url in case you want to subtly
change it and try again.

If you go to `/sars-cov-2/*`, any value of
`*` is assumed to be intended as a filter
you'd like to apply on the dataset listing.
E.g. `/sars-cov-2/neherlab` filters to
datasets matching the keyword `neherlab`.

This is just one part of a broader attempt
to behave consistently when a dataset
doesn't exist on nextstrain.org.
For more, see https://github.com/nextstrain/nextstrain.org/issues/355

### CODE REVIEW

File: static-site/src/sections/sars-cov-2-page.jsx [hunk 1]
```diff
@@ -132,6 +133,21 @@ class Index extends React.Component {
       catalogueDatasets: sarscov2Catalogue.datasets,
       filterParsed: false
     };
+    if (this.props.uri === "/sars-cov-2") {
```
Comment (jameshadfield) [inline static-site/src/sections/sars-cov-2-page.jsx:136]: [notes from zoom call yesterday] Let's remove the URL-based filtering for the time being, so this added code can be removed, and thus we only need `nonExistentDatasetName = this.props["*"]` set as component state here in the constructor.

File: static-site/src/sections/sars-cov-2-page.jsx [hunk 2]
```diff
@@ -183,8 +217,8 @@ class Index extends React.Component {
                   "FilterDisplay",
                   "ListDatasets"
                 ]}
-                urlDefinedFilterPath={this.props["*"]}
-                intendedUri={this.props.uri}
+                urlDefinedFilterPath={this.state.urlDefinedFilterPath}
```
Comment (jameshadfield) [inline static-site/src/sections/sars-cov-2-page.jsx:220]: [notes from zoom call yesterday] these props should be removed

File: static-site/src/sections/sars-cov-2-page.jsx [hunk 3]
```diff
@@ -143,9 +159,27 @@ class Index extends React.Component {
     }
   }
 
+  banner() {
+    let bannerTitle, bannerContents;
+    if (this.state.nonExistentDatasetName) {
+      bannerTitle = `The dataset "nextstrain.org/ncov/${this.state.nonExistentDatasetName}" doesn't exist.`;
```
Comment (jameshadfield) [inline static-site/src/sections/sars-cov-2-page.jsx:165]: [notes from zoom call yesterday] banner message:

The dataset "nextstrain.org${this.props.uri}/${this.state.nonExistentDatasetName}" doesn't seem to exist. Please <a href="https://discussion.nextstrain.org/">post on the discussion board</a> if you believe there has been an error here.

Below is our main SARS-CoV-2 page which lists all current datasets & narratives which we are aware of.

**Review by jameshadfield — CHANGES_REQUESTED** [review]

**Comment by eharkins** [comment]
@jameshadfield 669c953 is my first attempt at what we discussed this week. Let me know what you think!

File: static-site/src/sections/sars-cov-2-page.jsx [hunk 4]
```diff
@@ -135,17 +135,42 @@ class Index extends React.Component {
...
+  }
+
+  banner() {
+    if (this.state.nonExistentDatasetName && (this.state.nonExistentDatasetName.length > 0)) {
+      const bannerTitle = `The dataset "nextstrain.org${this.props.location.pathname}" doesn't exist.`;
+      const bannerContents = `Here is the SARS-CoV-2 page, where we have listed featured datasets,
+      narratives, and resources related to SARS-CoV-2. Note that some SARS-CoV-2
+      datasets may not be listed here. For a more comprehensive list of
+      Nextstrain-maintained (including SARS-CoV-2) datasets,
+      check out ${<a href="/pathogens">nextstrain.org/pathogens</a>}.`;
```
Comment (jameshadfield) [inline static-site/src/sections/sars-cov-2-page.jsx:164]: This is being rendered as 
```
check out [object Object].
```

Comment (eharkins) [reply static-site/src/sections/sars-cov-2-page.jsx]: Thanks @jameshadfield! Should be fixed in 8a2dc68.

**Review by jameshadfield — APPROVED** [review]

**Pull request merged**